tests/php/Unit/MollieWCHelperSettingsTest.php

```diff
@@ -0,0 +1,482 @@
+     */
+    public function testGetPaymentLocaleReturnsWpLocale()
+    {
+        /*
+         * Stubs
+         */
+        $validLanguageCode = 'en_US';
+
+        /*
+         * Setup Testee
+         */
+        list($testee, $testeeMethod) = $this->buildTesteeMethodMock(
+            Testee::class,
+            [],
+            'getPaymentLocale',
+            [
+                'getPaymentLocaleSetting',
+                'getCurrentLocale',
+            ]
+        );
+
+        /*
+         * Expect getPaymentLocaleSettings returns Testee::SETTING_LOCALE_WP_LANGUAGE_CODE
+         */
+        $testee
+            ->expects($this->once())
+            ->method('getPaymentLocaleSetting')
+            ->willReturn(Testee::SETTING_LOCALE_WP_LANGUAGE);
+
+        /*
+         * Then expect getCurrentLocale is called and return a valid language code value.
+         */
+        $testee
+            ->expects($this->once())
+            ->method('getCurrentLocale')
+            ->willReturn($validLanguageCode);
+
+        /*
+         * Execute test
+         */
+        $result = $testeeMethod->invoke($testee);
+
+        self::assertEquals($validLanguageCode, $result);
+    }
+
+    /**
+     * Test Default Browser language is returned
+     */
+    public function testGetPaymentLocaleReturnsDefaultBrowserLanguage()
+    {
+        /*
+         * Stubs
+         */
+        $validLanguageCode = 'en_US';
+
+        /*
+         * Setup Testee
+         */
+        list($testee, $testeeMethod) = $this->buildTesteeMethodMock(
+            Testee::class,
+            [],
+            'getPaymentLocale',
+            [
+                'getPaymentLocaleSetting',
+                'browserLanguage',
+            ]
+        );
+
+        /*
+         * Expect to call getPaymentLocaleSettings and return Testee::SETTING_LOCALE_BY_BROWSER
+         */
+        $testee
+            ->expects($this->once())
+            ->method('getPaymentLocaleSetting')
+            ->willReturn(Testee::SETTING_LOCALE_DETECT_BY_BROWSER);
+
+        /*
+         * Then expect to call browserLanguage to retrieve the browser language
+         */
+        $testee
+            ->expects($this->once())
+            ->method('browserLanguage')
+            ->willReturn($validLanguageCode);
+
+        /*
+         * Execute Test
+         */
+        $result = $testeeMethod->invoke($testee);
+
+        self::assertEquals($validLanguageCode, $result);
+    }
+
+    /* -----------------------------------------------------------------
+       getPaymentLocaleSetting Tests
+       -------------------------------------------------------------- */
+
+    /**
+     * Test Wp Language option is returned as default value if options from
+     * database is a falsy value.
+     */
+    public function testGetPaymentLocaleSettingsReturnWpLanguagePlaceholderAsDefaultValue()
+    {
+        /*
+         * Stubs
+         */
+        $settingId = uniqid(mt_rand(0, 8));
```

Comment:
Good job randomizing the ID. Although, why the need for additional prefix randomisation with `mt_rand()`?

Reply:
The mt_rand() function is a drop-in replacement for the older rand(). It uses a random number generator with known characteristics using the » Mersenne Twister, which will produce random numbers four times faster than what the average libc rand() provides.

https://www.php.net/manual/en/function.mt-rand.php

Reply:
I know what `mt_rand()` is :) I also understand why you need to add the prefix. I'm asking why the _randomization_ of the prefix. If anything, IMHO the prefix should be something meaningful, not random: the random part is already handled by `uniqid()`. If you think that the value generated by `uniqid()` is not unique enough, just use the second argument, which by the way is computationally cheaper.